node/narwhal/src/primary.rs

```diff
@@ -155,8 +155,19 @@ impl<N: Network> Primary<N> {
         // Sign the batch.
         let batch = Batch::new(private_key, round, transmissions, previous_certificates, &mut rng)?;
 
-        // Set the proposed batch.
-        *self.proposed_batch.write() = Some((batch.clone(), Default::default()));
+        // Generate a timestamp.
+        let timestamp = now();
+        // Sign the batch ID.
+        let signature =
```

Comment:
Our signature is already in the `Batch` struct. This step should be skipped.

Instead, when checking quorum threshold, one needs to pull out the signature there.

Reply:
I have fixed this in the `narwhal` branch here:
https://github.com/AleoHQ/snarkOS/commit/34813e31f6f226922b1f331b982384eabb839432

Reply:
Thanks for pointing this out

Reply:
Oh, if there's a better fix, sure. First thing I tried was shutting down one validator, and immediately the quorum stopped.